crates/rspack_fs/src/sync.rs

```diff
@@ -23,11 +30,18 @@ pub trait WritableFileSystem {
   fn write(&self, file: &Utf8Path, data: &[u8]) -> Result<()>;
 }
 
-pub trait ReadableFileSystem {
-  /// Read the entire contents of a file into a bytes vector.
-  ///
-  /// Error: This function will return an error if path does not already exist.
-  fn read(&self, file: &Utf8Path) -> Result<Vec<u8>>;
+pub trait ReadableFileSystem: Send + Sync {
+  /// See [std::fs::read]
+  fn read(&self, path: &Path) -> io::Result<Vec<u8>>;
+
+  /// See [std::fs::metadata]
+  fn metadata(&self, path: &Path) -> io::Result<FileMetadata>;
```

Comment:
I was a little bit confused. Why would this use structs defined in `rspack_resolver`, shouldn't this be a generic interface?

Reply:
you mean FileMetadata or FileSystem?

Reply:
`FileMetadata`, sorry for not being clear.

Reply:
yes use std FileMetadata makes more sense, I will fix it

Reply:
fixed